showkase-processor/src/main/java/com/airbnb/android/showkase/processor/models/ShowkaseMetadata.kt

```diff
@@ -261,10 +278,49 @@ internal fun getShowkaseMetadataFromPreview(
         showkaseHeightDp = map[ShowkaseAnnotationProperty.HEIGHTDP]?.let { it as Int },
         insideWrapperClass = showkaseFunctionType == ShowkaseFunctionType.INSIDE_CLASS,
         insideObject = showkaseFunctionType.insideObject(),
-        element = element
+        element = element,
+        previewParameter = previewParameterTypeMirror 
     )
 }
 
+private fun ExecutableElement.getPreviewParameterTypeMirror(
+    typeUtils: Types,
+    previewParameterTypeMirror: TypeMirror,
+    elementUtil: Elements
+): TypeMirror? {
+    val previewParametersMap =
+        processPreviewParameterAnnotation(typeUtils, previewParameterTypeMirror)
+            .firstOrNull()
+    val previewParameterTypeMirror =
+        previewParametersMap?.get(PreviewParameterProperty.PROVIDER)?.let {
+            elementUtil.getTypeElement(it.toString()).asType()
+        }
+    return previewParameterTypeMirror
+}
+
+private fun ExecutableElement.processPreviewParameterAnnotation(
+    typeUtils: Types,
+    previewParameterTypeMirror: TypeMirror
+) = parameters
+    .map {
+        val map = mutableMapOf<PreviewParameterProperty, Any>()
+        val previewParameterAnnotation = it.annotationMirrors.find {
+            typeUtils.isSameType(it.annotationType, previewParameterTypeMirror)
+        }
+        previewParameterAnnotation?.elementValues?.map { entry ->
```

Comment:
I think you can use one of the `associate` std lib functions to avoid manually managing a mutable map? (could filter PreviewParameterProperty.values() first).